File: height_tree/height_tree.py
class Node:
    def __init__(self, val, arrayNodes):
        self.val = val
        self.nodes = {}

node1 = Node(12, {node2, node3})
node2 = Node(99, [node4])
node3 = Node(30, [node5])
node4 = Node(30, [node7, node8])
node5 = Node(30, [node9])
node6 = Node(30)
node7 = Node(30)
node8 = Node(30)
node9 = Node(30)

def height_tree(root):
    if root is None:
        return -1
    else:
        i = 0
        nodeCount = len(root.nodes)
        max = 0
        while i < nodeCount:
            if height_tree(root.nodes[i]) > max:
                max = height_tree(root.nodes[i])
            i += 1
        return max + 1

print(height_tree(node1))

File: height_tree/height_tree.c
#include <stdio.h>
#include <stdlib.h>

struct s_node {
        int           value;
        struct s_node **nodes;
};

void	traverse_tree(struct s_node *root, int curr, int *max)
{
	int i = 0;

	if (!root)
		return;
	if (curr > *max)
		*max = curr;
	while (root->nodes[i])
	{
		traverse_tree(root->nodes[i], curr + 1, max);
		i++;
	}
}

int     height_tree(struct s_node *root)
{
    int curr = 0;
    int max = 0;
    traverse_tree(root, curr, &max);
	return (max);
}
/*
#include <stdio.h>
#include <stdlib.h>

struct s_node* b(int v) {
    struct s_node* new = malloc(sizeof(new));
    new->value = v;
    new->nodes = malloc(100000);
    return new;
}

int main() {

    struct s_node* r = b(94);
    r->nodes[0] = b(34);
    r->nodes[1] = b(52);

    r->nodes[0]->nodes[0] = b(1);
    r->nodes[0]->nodes[1] = b(99);
    r->nodes[0]->nodes[2] = b(11);

    r->nodes[0]->nodes[1]->nodes[0] = b(13);

    printf("%d\n", height_tree(r));

    return 0;
}
*/
